### PR TITLE
Update the Starter's Guide

### LINKED ISSUE #273: Update the Starter's Guide
The builder's starter guide is now out-of-date, it should be brought up to modern times.

### PR DESCRIPTION
**Pull request by lsaether**
closes #273

### CODE REVIEW

File: docs/build-build-with-polkadot.md [hunk 1]
```diff
@@ -19,33 +19,38 @@ that you can get started creating your application in anticipation of the Polkad
...
+2.0.0 tagged release that will solidify its API.
+
+Polkadot does not natively support smart contracts, however there will be parachains
+that do. Substrate chains can include smart contract functionality by using the provided 
+[contracts Frame pallet](https://github.com/paritytech/substrate/tree/master/frame/contracts).
+The contracts pallet enables a chain to use Wasm-compiled contracts that can be deployed
+permissionlessly by users or with specific rules dependent on the chain. 
+To facilitate development of Wasm smart contracts, Parity is also developing 
+[ink!](https://github.com/paritytech/ink), a domain specific language built
+in Rust for writing smart contracts.
```
Comment (Swader) [inline docs/build-build-with-polkadot.md]: Let's mention the EVM pallet here as well?

**Review by Swader — APPROVED** [review]
While we're editing this, we should probably address the new marketing speak as well, i.e. remove "shared security" or de-emphasize it and replace "interoperability" with "composeability".

File: docs/build-build-with-polkadot.md (hunk 1 above)
Comment (lsaether) [reply docs/build-build-with-polkadot.md]: Done

**Comment by lsaether** [comment]
Let's do the new marketing speak in a separate PR. See #281

**Pull request merged**